File: 231-power-of-two/231-power-of-two.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfTwo = function(n) {
  let count=0;
  if(n<0){
      return 0;
  }
  while(n != 0){
      n=n&n-1;
      count++;
  }
    return count == 1;
};

File: 231-power-of-two/231-power-of-two.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        int count=0;
        if(n < 0){
            return false;
        }
        while(n != 0){
            n = n & (n-1);
            count++;
        }
        if(count == 1){
            return true;
        }
        return false;
    }
};